pkg/haprovider/haprovider.go

```diff
@@ -123,6 +122,45 @@ func (r *HAProvider) createService(
 	return service, err
 }
 
+func (r *HAProvider) getServiceAnnotation(ctx context.Context, cluster *clusterv1.Cluster) map[string]string {
+	serviceAnnotation := map[string]string{
+		akoov1alpha1.HAServiceAnnotationsKey:  "true",
+		akoov1alpha1.TKGClusterNameLabel:      cluster.Name,
+		akoov1alpha1.TKGClusterNameSpaceLabel: cluster.Namespace,
+	}
+
+	aviInfraSetting := &akov1alpha1.AviInfraSetting{}
+
+	// TODO: check a cluster should be managed by only one adc
+	adcForCluster, err := handlers.GetADCForCluster(ctx, cluster, r.log, r.Client)
+	if err != nil {
+		return serviceAnnotation
+	}
+
+	if len(adcForCluster) == 0 {
+		r.log.Error(err, "Failed to get akoDeploymentConfig for current cluster")
+		return serviceAnnotation
+	}
+
+	aviInfraSettingName := GetAviInfraSettingName(&adcForCluster[0])
```

Comment:
As I said before, you should abstract this to another function and call it here, don't add too many logic in one function.

Reply:
getServiceAnnotation() is the function I abstracted from createService() based on your previous comment. Do you think I should abstracte another function from getServiceAnnotation()?

Reply:
- It is better to name this function: `annotateService`
- I think you can abstract the avi infra setting related logic to a new function like this (maybe a better name):
```go
func(r *HAProvider) findServiceBelongedAIS(*akov1alpha1.AviInfraSetting)
```